src/opentime/timeRange.h

```diff
@@ -82,6 +82,34 @@ class TimeRange {
         return _start_time < other.end_time_exclusive() && other._start_time < end_time_exclusive();
     }
 
+    bool before(TimeRange other) const {
+        return _start_time < other._start_time && end_time_exclusive() <= other._start_time;
```

Comment:
This is a question for the team at large. Is there any reason why a time range could ever have a start time greater than then end time? If no, through this file, there are comparisons such as this one where the _start_time < other._start_time is redundant with the subsequent check and can be eliminated. If the start IS allowed be greater than the end, then this entire file needs to be scrubbed so that all permutations of unordered TimeRanges are accommodated. If it is disallowed, then we also need code that enforces end >= start.

Reply:
This isn't something I've tested. How is reversing of clips handled in different NLEs n OTIO? Afair in premiere we can change the speed to -100%. So I guess that would be exported as a time warp in OTIO? In this case we won't need to allow greater start times. But this might be different for different software packages.
Other than reversing a clip, I can't think of any other use case for this.

Reply:
You're right, a timewarp is the right answer for reversing. TimeRanges should never run backwards on their own.